src/reducer.js

```diff
@@ -38,24 +40,72 @@ const pathToArr = path => path.split(/\//).filter(p => !!p)
  * @return {Map} State
  */
 export default (state = initialState, action = {}) => {
-  const { path } = action
+  const { path, timestamp, requesting, requested } = action
   let pathArr
+  let rootPathArr
   let retVal
 
   switch (action.type) {
 
+    case START:
+      pathArr = pathToArr(path)
+      retVal = (requesting !== undefined)
+          ? state.setIn(['requesting', ...pathArr], fromJS(requesting))
+          : state.deleteIn(['requesting', ...pathArr])
+
+      retVal = (requested !== undefined)
+          ? retVal.setIn(['requested', ...pathArr], fromJS(requested))
+          : retVal.deleteIn(['requested', ...pathArr])
+
+      return retVal;
+
     case SET:
-      const { data } = action
+      const { data, rootPath } = action
       pathArr = pathToArr(path)
+      rootPathArr = pathToArr(rootPath)
       retVal = (data !== undefined)
         ? state.setIn(['data', ...pathArr], fromJS(data))
         : state.deleteIn(['data', ...pathArr])
 
+      retVal = (timestamp !== undefined)
+        ? retVal.setIn(['timestamp', ...rootPathArr], fromJS(timestamp))
```

Comment:
Is there a time where timestamp will ever not be `Date.now()`? If so maybe we should just have this line be `retVal.setIn(['timestamp', ...rootPathArr], fromJS(Date.now()))` and we wouldn't have to pass timestamp into the actions.